examples/stable_diffusion_xl/tests/test_modules/BasicTransformerBlock_forward.py

```diff
@@ -0,0 +1,193 @@
+import argparse
+import ast
+import os
+
+import numpy as np
+
+import mindspore as ms
+import mindspore.dataset as de
+from mindspore import Profiler, nn
+
+
+def create_loader(
+    total_batch_size,
+    size=(),
+    dtypes=None,
+    num_parallel_workers=1,
+    shuffle=True,
+    drop_remainder=True,
+    python_multiprocessing=False,
+    seed=1,
+    dataset_column_names=["data"],
+):
+    dataset = Dataset(size=size, dtypes=dtypes)
+
+    de.config.set_seed(seed)
+    ds = de.GeneratorDataset(
+        dataset,
+        column_names=dataset_column_names,
+        num_parallel_workers=min(8, num_parallel_workers),
+        shuffle=shuffle,
+        python_multiprocessing=python_multiprocessing,
+    )
+    per_batch_size = total_batch_size
+
+    ds = ds.batch(
+        per_batch_size,
+        drop_remainder=drop_remainder,
+    )
+    ds = ds.repeat(1)
+
+    return ds
+
+
+class Dataset:
```

Comment:
Too many duplicated class implementation for Dataset. 
For those classes and functions are commonly used by all test scripts, please put them into a file like  `common.py`

Reply:
Done